### ISSUE TITLE
unable to insert data with foreign_key to enum table

### ISSUE THREAD

**Issue by polRk** [hasura/graphql-engine#3808]
I tried to set the value for the enum of the linked column

```log
main.js.gz:1 Uncaught (in promise) TypeError: Cannot use 'in' operator to search for 'info' in Internal Server Error
    at n (main.js.gz:1)
    at n (main.js.gz:1)
    at main.js.gz:1
```

Https query 
```json
{"type":"update","args":{"table":{"schema":"public","name":"_action"},"$set":{"type":"viber_bot","description":null},"$default":[],"where":{"id":"d54869a7-77b5-4fa4-b8cb-8707bca172fa"}}}
```

It helped me to make the table normal from enum first, and then make enum again

**Comment by rikinsk** [comment]
@polRk Can you share some more information on what exactly you were attempting to do. A little more info about your schema and the action you were attempting will help

**Comment by polRk** [comment]
My table 

<img width="409" alt="Снимок экрана 2020-01-31 в 4 17 34 PM" src="https://user-images.githubusercontent.com/39828645/73542185-345a8080-4445-11ea-8f64-61e268c327ce.png">

My additional table options
<img width="552" alt="Снимок экрана 2020-01-31 в 4 18 00 PM" src="https://user-images.githubusercontent.com/39828645/73542253-55bb6c80-4445-11ea-99f9-44be5ca9815d.png">

My enum table _action_type
<img width="366" alt="Снимок экрана 2020-01-31 в 4 19 37 PM" src="https://user-images.githubusercontent.com/39828645/73542309-7be10c80-4445-11ea-84b0-3df8bd81f151.png">

If I am not mistaken when I created the connection, the action for updating Foreign Key - On Update Violation: restrict. After which I updated and deleted some values ​​in my enum table. And when adding a new record to the data table, there was an error. I recreated the enum table with toggle button and Make On Update Violation: cascade. And it worked

**Comment by rikinsk** [comment]
Thanks for the extra information. I unfortunately still did not understand the actual issue. This is what I have understood till now:

- You have an enum table `_action_type`
- You have a table `_action` that has a foreign key to the enum table.
- You already had some data in the `_action` table and then you added/edited some data in the `_action_type` table. 
- Now while adding data to the `_action` table you faced an error. Was the value set for `type` column was not present in the `_action_type` table (causing a foreign-key violation)? Or was the error due to something else?
- The 'Internal Server Error' msg changes to a more meaningful error if the table is not marked as an enum table which helped you resolve your issue. This a bug that needs to be fixed.

Am I missing something else here?

**Comment by polRk** [comment]
> Now while adding data to the _action table you faced an error. Was the value set for type column was not present in the _action_type table (causing a foreign-key violation)? Or was the error due to something else?

It is present. I deleted and created the value several times, but the error remained.

> The 'Internal Server Error' msg changes to a more meaningful error if the table is not marked as an enum table which helped you resolve your issue. This a bug that needs to be fixed.

I think the problem was that the database did not understand that it was enum. After removing the enum label and turning it on the enum label, it worked

**Comment by lexi-lambda** [comment]
@polRk Is this still an issue? I’m unclear on whether or not it’s been resolved.

**Comment by polRk** [comment]
I could not reproduce. But yes, there is a problem with enum

**Comment by lexi-lambda** [comment]
It’s possible that the issue was fixed by #3798, which made it into `1.1.0-beta.3`. If you can’t reproduce the issue, I’m going to close it, but feel free to reopen it (or open a new issue) if it happens again.
